index.js

```diff
@@ -1,443 +0,0 @@
-
-  // Transform CSV into an array
-  let contexts = csvparse(contextsCsv, {
-    columns: ['viewport', 'density', 'views'],
-    from: csvHasHeader ? 2 : 1,
-    cast: function(value, context) {
-      if (context.column == 'density') {
-        return parseFloat(value)
-      } else {
-        return parseInt(value, 10)
-      }
-    },
-  })
-  if (argv.verbose) {
-    console.log(color.green(`Imported ${contexts.length} lines of context`))
-  }
-  const contextMinViewport = contexts.reduce(
-    (min, p) => (p.viewport < min ? p.viewport : min),
-    contexts[0].viewport,
-  )
-  const contextMaxViewport = contexts.reduce(
-    (max, p) => (p.viewport > max ? p.viewport : max),
-    contexts[0].viewport,
-  )
-  if (argv.verbose) {
-    console.log(
-      color.green(
-        `Viewports in context go from ${contextMinViewport}px to ${contextMaxViewport}px`,
-      ),
-    )
-  }
-
-  minViewport = contextMinViewport
-  if (argv.minviewport !== undefined) {
-    minViewport = Math.max(contextMinViewport, argv.minviewport)
-  }
-  maxViewport = contextMaxViewport
-  if (argv.maxviewport !== undefined) {
-    maxViewport = Math.min(contextMaxViewport, argv.maxviewport)
-  }
-
-  if (argv.verbose) {
-    console.log(
-      color.green(
-        `Viewports will be considered from ${color.white(
-          minViewport + 'px',
-        )} to ${color.white(maxViewport + 'px')}`,
-      ),
-    )
-  }
-
-  /* ======================================================================== */
-  if (argv.verbose) {
-    console.log(
-      color.bgCyan.black(
-        '\nStep 2: get variations of image width across viewport widths',
-      ),
-    )
-  }
-
-  const VIEWPORT = {
-    width: minViewport,
-    height: 2000,
-    deviceScaleFactor: 1,
-  }
-  const imageWidths = []
-  if (argv.verbose) {
-    console.log(color.green('Launch headless Chrome'))
-  }
-  const browser = await puppeteer.launch()
-  const page = await browser.newPage()
-  if (argv.verbose) {
-    console.log(color.green(`Go to ${argv.url}`))
-  }
-  await page
-    .goto(argv.url, { waitUntil: 'networkidle2' })
-    .then(async () => {
-      if (argv.verbose) {
-        console.log(
-          color.green(`Checking widths of image ${color.white(argv.selector)}`),
-        )
-        process.stdout.write(
-          `Current viewport: ${color.cyan(VIEWPORT.width)}px`,
-        )
-      }
-      while (VIEWPORT.width <= maxViewport) {
-        // Set new viewport width
-        await page.setViewport(VIEWPORT)
-
-        // Give the browser some time to adjust layout, sometimes requiring JS
-        await sleep(argv.delay)
-
-        // Check image width
-        let imageWidth = await page.evaluate(sel => {
-          return document.querySelector(sel).width
-        }, argv.selector)
-        imageWidths[VIEWPORT.width] = imageWidth
-
-        // Increment viewport width
-        VIEWPORT.width++
-
-        // Update log in the console
```

Comment:
@ryuran I took this back in PR #43 to get a nicer log with viewport width increasing on a single line